### PR TITLE
Offline CSV generation

### PR DESCRIPTION
**Pull request by dracos**
Fixes https://github.com/mysociety/fixmystreet-commercial/issues/1926

From the command line, can call with export=2 instead of 1 to get the 202 flow, otherwise is as before.

### CODE REVIEW

File: t/app/controller/dashboard.t [hunk 1]
```diff
@@ -252,7 +259,30 @@ FixMyStreet::override_config {
...
+        (my $name = $f->basename) =~ s/-part$//;;
+        $mech->log_in_ok( $counciluser->email );
+        $mech->get_ok('/dashboard/status');
+        $mech->content_contains('/dashboard/csv/www.example.org-body-' . $body->id . '-start_date-2014-01-02.csv');
+        $mech->content_like(qr/$name\s*<br>0KB\s*<i>In progress/);
+
+        my $token = 'access_token=' . $counciluser->id . '-1234567890abcdefgh';
+        $mech->get_ok("/dashboard/csv/$name?$token");
+        is $mech->res->code, 202;
+    }
```
Comment (struan) [inline t/app/controller/dashboard.t:285]: Should we not also check that if all files are generated it reflects that too. If for no other reason that to check we don't get endless request loops if someone leaves the tab open.

File: perllib/FixMyStreet/Reporting.pm [hunk 2]
```diff
@@ -305,6 +306,46 @@ sub generate_csv {
 
 # Output code
 
+sub cache_dir {
+    my $self = shift;
+
+    my $cfg = FixMyStreet->config('PHOTO_STORAGE_OPTIONS');
+    my $dir = $cfg ? $cfg->{UPLOAD_DIR} : FixMyStreet->config('UPLOAD_DIR');
+    $dir = path($dir, "dashboard_csv")->absolute(FixMyStreet->path_to());
+    my $subdir = $self->user ? $self->user->id : 0;
```
Comment (struan) [inline perllib/FixMyStreet/Reporting.pm:315]: Should there not always be a user here?

Comment (dracos) [reply perllib/FixMyStreet/Reporting.pm:315]: No, kiitc uses the cobrand hook to allow non-authenticated access to the CSV export.

Comment (dracos) [reply perllib/FixMyStreet/Reporting.pm:315]: Though they will still be using export=1 and not be large, so wouldn't hit this. But still, might as well be safe.

**Pull request merged**